src/java/org/apache/cassandra/tools/nodetool/GetCompactionThroughput.java

```diff
@@ -17,17 +17,34 @@
  */
 package org.apache.cassandra.tools.nodetool;
 
+import com.google.common.math.DoubleMath;
+
 import io.airlift.airline.Command;
 
+import io.airlift.airline.Option;
 import org.apache.cassandra.tools.NodeProbe;
 import org.apache.cassandra.tools.NodeTool.NodeToolCmd;
 
-@Command(name = "getcompactionthroughput", description = "Print the MiB/s throughput cap for compaction in the system")
+@Command(name = "getcompactionthroughput", description = "Print the MiB/s throughput cap for compaction in the system as a rounded number")
 public class GetCompactionThroughput extends NodeToolCmd
 {
+    @SuppressWarnings("UnusedDeclaration")
+    @Option(name = { "-d", "--precise-mib" }, description = "Print the MiB/s throughput cap for compaction in the system as a precise number (double)")
+    private boolean  compactionThroughputAsDouble;
```

Comment:
It's unfortunate that existing tooling might be expecting an integer here, but I guess this is the best trade-off we can make for now. Perhaps we can deprecate the integral return value and just remove it in 5.0...

Reply:
Yeah, I will also document all this today in NEWS.txt and probably in other places to be more noisy for people to consider it and I hope we can get rid of it in 5.0